bentoml/deployment/aws_lambda/utils.py

```diff
@@ -0,0 +1,382 @@
+bento_service = bento_service_cls()
+bento_service._load_artifacts('/tmp/bentoml')
+
+"""
+
+AWS_FUNCTION_TEMPLATE = """\
+def {api_name}(event, context):
+    api = bento_service.get_service_api('{api_name}')
+
+    return api.handle_aws_lambda_event(event)
+
+"""
+
+
+def ensure_sam_available_or_raise():
+    try:
+        import samcli  # noqa
+    except ImportError:
+        raise ImportError(
+            'aws-sam-cli package is required. Install '
+            'with `pip install --user aws-sam-cli`'
+        )
+
+
+def cleanup_build_files(project_dir, api_name):
+    build_dir = os.path.join(project_dir, '.aws-sam/build/{}'.format(api_name))
+    logger.debug('Cleaning up unused files in SAM built directory {}'.format(build_dir))
+    if which('du'):
+        logger.debug(subprocess.check_output(['du', '-sch'], cwd=build_dir))
+    try:
+        for root, dirs, files in os.walk(build_dir):
+            if 'tests' in dirs:
+                logger.debug('removing dir: ' + os.path.join(root, 'tests'))
+                shutil.rmtree(os.path.join(root, 'tests'))
+            if 'test' in dirs:
+                logger.debug('removing dir: ' + os.path.join(root, 'test'))
+                shutil.rmtree(os.path.join(root, 'test'))
+            if 'examples' in dirs:
+                logger.debug('removing dir: ' + os.path.join(root, 'examples'))
+                shutil.rmtree(os.path.join(root, 'examples'))
+            if '__pycache__' in dirs:
+                logger.debug('removing dir: ' + os.path.join(root, '__pycache__'))
+                shutil.rmtree(os.path.join(root, '__pycache__'))
+
+            for dir_name in filter(lambda x: re.match('^.*\\.dist-info$', x), dirs):
+                logger.debug('removing dir ' + dir_name)
+                shutil.rmtree(os.path.join(root, dir_name))
+
+            for file in filter(lambda x: re.match('^.*\\.egg-info$', x), files):
+                logger.debug('removing file: ' + os.path.join(root, file))
+                os.remove(os.path.join(root, file))
+
+            for file in filter(lambda x: re.match('^.*\\.pyc$', x), files):
+                logger.debug('removing file: ' + os.path.join(root, file))
+                os.remove(os.path.join(root, file))
+
+            if 'caff2' in files:
+                logger.debug('removing file: ' + os.path.join(root, 'caff2'))
+                os.remove(os.path.join(root, 'caff2'))
+            if 'wheel' in files:
+                logger.debug('removing file: ' + os.path.join(root, 'wheel'))
+                os.remove(os.path.join(root, 'wheel'))
+            if 'pip' in files:
+                logger.debug('removing file: ' + os.path.join(root, 'pip'))
+                os.remove(os.path.join(root, 'pip'))
+            if 'libtorch.so' in files:
+                logger.debug('removing file: ' + os.path.join(root, 'libtorch.so'))
+                os.remove(os.path.join(root, 'libtorch.so'))
+    except Exception as e:
```

Comment:
why do we need this?

Reply:
To shrink down the pytorch deployment size.  I am going to remove it for now, until we have more testings around it

Reply:
I was asking why " except Exception as e"